app/src/server/utils/stripe.ts

```diff
@@ -0,0 +1,144 @@
+import Stripe from "stripe";
+import { env } from "~/env.mjs";
+import { prisma } from "~/server/db";
+import { TRPCError } from "@trpc/server";
+import type { Decimal } from "@prisma/client/runtime";
+
+const stripe = new Stripe(env.STRIPE_SECRET_KEY ?? "");
+
+export async function createStripePaymentIntent({
+  amount,
+  stripeCustomerId,
+  paymentMethodId,
+  returnUrl,
+  invoiceId,
+}: {
+  amount: number;
+  stripeCustomerId: string;
+  paymentMethodId: string;
+  returnUrl: string;
+  invoiceId: string;
+}) {
+  return stripe.paymentIntents.create({
+    amount,
+    currency: "usd",
+    automatic_payment_methods: { enabled: true },
+    customer: stripeCustomerId,
+    payment_method: paymentMethodId,
+    return_url: returnUrl,
+    off_session: true,
+    confirm: true,
+    metadata: {
+      invoiceId, // Used to track status using webhooks
+    },
+  });
+}
+export async function createSetupIntent(stripeCustomerId: string) {
```

Comment:
You could make these functions even more streamlined with arrow function syntax:

```
export const createSeupIntent = async (stripeCustomerId: string) =>
stripe.setupIntents.create({
...
```

Reply:
Yes, it is a good place for an arrow function.